lib/index.js

```diff
@@ -89,44 +89,102 @@ AdWords.prototype.page = function(page) {
  */
 
 AdWords.prototype.track = function(track) {
-  var id = this.options.conversionId;
+  var self = this;
   var props = track.properties();
   var remarketing = this.options.remarketing;
-  var events = this.events(track.event());
+  var labelMap = this.options.labelMap;
   var revenue = track.revenue() || 0;
-  // Check if this is a whitelisted event for standalone remarketing tag
-  var whitelisted = this.options.whitelist.indexOf(track.event()) > -1;
-  var sentAlready = false;
+  var remarketingSent = {}; // for each conversion id tracks whether a remarketing tag has been sent yet
 
-  each(function(label) {
-    delete props.revenue;
-    // Fire conversion tag
-    window.google_trackConversion({
-      google_conversion_id: id,
-      google_custom_params: props,
-      google_conversion_language: 'en',
-      google_conversion_format: '3',
-      google_conversion_color: 'ffffff',
-      google_conversion_label: label,
-      google_conversion_value: revenue,
-      google_remarketing_only: false // ensure this is a conversion tag
+  if (labelMap) { // if labelmap exists then the new metadata schema is being used
+    labelMap.forEach(function(mapping) {
+      if (typeof mapping.value.eventName !== 'string' || typeof track.event() !== 'string') return;
```

Comment:
this is probably unnecessary since it'd impossible for them to not set a string value. The component setting for eventName would be `string` type so it would always be stored as a string in our DB. track.event() is always a string